pkgs/servers/web-apps/vikunja/frontend.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, lib, fetchFromGitea, mkYarnPackage, yarn, apiBase ? "/api/v1", ... }:
+
+let
+  version = "0.17.0";
+
+  src = fetchFromGitea {
+    domain = "kolaente.dev";
+    owner = "vikunja";
+    repo = "frontend";
+    rev = "v${version}";
+    sha256 = "sha256-/KCpsPEhCJz+1pLisSviFP/VCiXUD6N9l1ZragOd2Go=";
+  };
+  frontend-modules = mkYarnPackage rec {
+    name = "vikunja-frontend-modules";
+    inherit version src;
+
+    packageJSON = ./frontend-package.json;
+    yarnLock = ./frontend-yarn.lock;
+    yarnNix = ./frontend-yarn.nix;
+
+    doDist = false;
+  };
+in stdenv.mkDerivation {
+  pname = "vikunja-frontend";
+  inherit version src;
+
+  nativeBuildInputs = [ frontend-modules yarn ];
+
+  buildPhase = ''
+    # Cannot use symlink or postcss-loader will crap out
+    cp -r ${frontend-modules}/libexec/vikunja-frontend/node_modules/ .
+    yarn --offline run build
+    # Unfortunately, this needs to be hardcoded at build.
```

Comment:
Actually, this is only kind of true: You can configure the api url at runtime in the frontend itself, it will only be pre-filled with the value set like this. If the variable is empty it will show the input without any prompts directly. We're [doing this for the desktop app](https://kolaente.dev/vikunja/desktop/src/branch/main/.drone.yml#L49).

Reply:
What do you think I should do? I think that is the best solution (with maybe changing the comment).

Reply:
I think this is fine. It could be changed later on, but for now it is fine.